From 37a1de7bcb3a831c01dd01a145a26a516e54bca7 Mon Sep 17 00:00:00 2001
From: Nikita Travkin <nikita@trvn.ru>
Date: Sat, 14 Jan 2023 12:07:31 +0500
Subject: [PATCH 4/4] REVERTME: arm64: dts: qcom: msm8916-longcheer-l8150:
 Enable the modem

This is a temporary change to enable the modem with minimal changes.
A different solution with a special msm8916-modem-sound.dtsi is planned
to be used upstream to handle the fact that the modem remoteproc acts as
a system adsp and takes over the audio hw upon boot.

Signed-off-by: Nikita Travkin <nikita@trvn.ru>
---
 .../boot/dts/qcom/msm8916-longcheer-l8150.dts  | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/msm8916-longcheer-l8150.dts b/arch/arm64/boot/dts/qcom/msm8916-longcheer-l8150.dts
index 9bdbdd506ed5..b15f54dc88da 100644
--- a/arch/arm64/boot/dts/qcom/msm8916-longcheer-l8150.dts
+++ b/arch/arm64/boot/dts/qcom/msm8916-longcheer-l8150.dts
@@ -9,6 +9,7 @@
 #include <dt-bindings/leds/common.h>
 
 /delete-node/ &wcnss_mem;
+/delete-node/ &mpss_mem;
 
 / {
 	model = "Longcheer L8150";
@@ -29,6 +30,11 @@ wcnss_mem: wcnss@8b600000 {
 			reg = <0x0 0x8b600000 0x0 0x600000>;
 			no-map;
 		};
+
+		mpss_mem: mpss@8ec00000 {
+			reg = <0x0 0x8ec00000 0x0 0x5000000>;
+			no-map;
+		};
 	};
 
 	gpio-keys {
@@ -76,6 +82,14 @@ flash_led: led {
 	};
 };
 
+&bam_dmux {
+	status = "okay";
+};
+
+&bam_dmux_dma {
+	status = "okay";
+};
+
 &blsp_i2c1 {
 	status = "okay";
 
@@ -218,6 +232,10 @@ &blsp1_uart2 {
 	status = "okay";
 };
 
+&mpss {
+	status = "okay";
+};
+
 &pm8916_resin {
 	status = "okay";
 	linux,code = <KEY_VOLUMEDOWN>;
-- 
2.38.1

